### PR TITLE
Create a DiningCommonsService and related controller method for fetching the names of all dining commons

### LINKED ISSUE #44: Subissue: Backend 1 - Dining commons service
Inherits from #39 

Following the code in proj-courses as a model, create a service that uses this api https://developer.ucsb.edu/apis/dining/dining-commons to get a list of all of the dining commons. Then build an endpoint: `GET /api/diningcommons/all` that will return all of the records returned by this endpoint.

### PR DESCRIPTION
**Pull request by Va1a**
Closes #44 
Dokku Deployment (for #56, #61, #62, #63): https://pd-consolidated-1.dokku-14.cs.ucsb.edu/

This resolves a subissue which is part of the epic "Home page lists dining commons and has a link to each"

In this PR, we implement DiningCommonsService.java , a Springboot service file that as of now allows us to access the UCSB dining commons API to fetch data from the following endpoint: /names. This endpoint returns the names and codes of all the dining commons at UCSB.
**Additionally, I added the following line to the application properties file:**
`app.ucsb.api.consumer_key=${UCSB_API_KEY:${env.UCSB_API_KEY:see-instructions-in-readme}}`
This allows us to define the UCSB_API_KEY in the .env file, so that it can be used within the application. This also means that the key NEEDS to be defined in the .env file (or as an env var) for proper functionality.

I also made a test file for DiningCommonsService. This gives us full test coverage and mutation coverage for the Service class implementation.
Finally, I created the DiningCommonsController file and added the following GET endpoint: "/api/diningcommons" which returns the name of all the dining commons as returned by the UCSB api.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/dining/controllers/DiningCommonsController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/diningcommons")
+public class DiningCommonsController {
+  private ObjectMapper mapper = new ObjectMapper();
+
+  @Autowired UserRepository userRepository;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/dining/controllers/DiningCommonsController.java:19]: Why do you need the UserRepository in this controller?  If you don't need it, remove it.

File: src/main/java/edu/ucsb/cs156/dining/controllers/DiningCommonsController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/diningcommons")
+public class DiningCommonsController {
+  private ObjectMapper mapper = new ObjectMapper();
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/dining/controllers/DiningCommonsController.java:17]: Are we using the mapper anywhere?  If not, consider removing it.

**Review by pconrad — CHANGES_REQUESTED** [review]
A couple of lines that you might consider removing.  Other than that, looks really good.

**Review by amanpdesai — APPROVED** [review]
Looks good to me!

**Review by amanpdesai — DISMISSED** [review]
Looks good to me!

**Review by pconrad — APPROVED** [review]
Code LGTM, but there is no dokku deployment on which to test.  Please fix that.

**Comment by pconrad** [comment]
On this PR: I'm testing on https://pd-consolidated-1.dokku-14.cs.ucsb.edu/ and two of the three new endpoints are working, while one is not working.

Is there something I'm doing wrong?

# Details

## The ones that work

`GET https://pd-consolidated-1.dokku-14.cs.ucsb.edu/api/diningcommons/all` works:

image

`GET https://pd-consolidated-1.dokku-14.cs.ucsb.edu/api/diningcommons/2024-11-24T00%3A00%3A00/de-la-guerra` works:

image

## The one that's not working

But the third one, the one that gets the menu items, is not working for me.  Am I just using bad values?

`GET https://pd-consolidated-1.dokku-14.cs.ucsb.edu/api/diningcommons/2024-11-24T00%3A00%3A00/dinner/de-la-guerra`

image

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**